traitsui/tests/editors/test_text_editor.py

```diff
@@ -0,0 +1,106 @@
+# ------------------------------------------------------------------------------
+#
+#  Copyright (c) 2020, Enthought, Inc.
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# ------------------------------------------------------------------------------
+import contextlib
+import unittest
+
+from traits.api import (
+    HasTraits,
+    Str,
+)
+from traitsui.api import TextEditor, View, Item
+from traitsui.tests._tools import (
+    GuiTestAssistant,
+    skip_if_not_qt4,
+    no_gui_test_assistant,
+)
+
+
+class Foo(HasTraits):
+
+    name = Str()
+
+
+def default_view():
+    return View(
+        Item(
+            name="name",
+        )
+    )
+
+
+def view_with_placeholder():
+    return View(
+        Item(
+            name="name",
+            editor=TextEditor(placeholder="Enter name"),
+        )
+    )
+
+
+def view_with_readonly_and_placeholder():
+    return View(
+        Item(
+            name="name",
+            editor=TextEditor(
+                placeholder="Enter name",
+                read_only=True,
+            )
+        )
+    )
+
+
+@contextlib.contextmanager
+def launch_ui(gui_test_case, object, view):
+    ui = object.edit_traits(view=view)
+    try:
+        yield ui
+    finally:
+        with gui_test_case.delete_widget(ui.control):
+            ui.dispose()
+
+
+# Skips tests if the backend is not either qt4 or qt5
+@skip_if_not_qt4
```

Comment:
Orthogonal: this always confuses me that this `skip_if_not_qt4` is actually `skip_if_not_qt`